doc/aaa/RBAC_Phase2.md

```diff
@@ -0,0 +1,421 @@
+    * *resource* : The xpath associated with the resource being accessed. This is a string.
+    * *tenant* : The tenant for which the resource partitioning is being done. This is a string.
+    * *instances* : The instances of the *resource* allocated to this *tenant*. This is a list of instances.
+  The TenantTable is keyed on <***resource, tenant***>
+
+### 3.2.2 APP DB
+N/A
+
+### 3.2.3 STATE DB
+N/A
+
+### 3.2.4 ASIC DB
+N/A
+
+### 3.2.5 COUNTER DB
+N/A
+
+### 3.2.6 USER DB
+This was described in the initial document for Phase 1, but the corresponding
+tables have been moved to CONFIG DB. This is therefore, no longer used.
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+To facilitate the sync between users in the UserDB and with users in Linux, a service must run on the host that listens for both changes to `/etc/passwd` and changes to UserDB, since users can be created via either interface (UserDB via NBIs / Linux commands).
+
+This service runs a process that uses the POSIX [inotify APIs](http://man7.org/linux/man-pages/man7/inotify.7.html) to register for file system events like changes to `/etc/passwd` and/or `/etc/shadow`.
+Another approach would be to have a process started by `systemd` on changes to `/etc/passwd` or `/etc/group`, and that process would simply reconcile the Redis DB with what is found in those files. `systemd` allows starting processes based on file create/delete/modify.
+
+A user can be created either via CLI or REST. It is the responsibility of this service to ensure that when the user information is added to the User DB, the appropriate user and groups are also created in the `/etc/passwd` and `/etc/groups` files.
+This way, the User DB information and the Linux groups information is always in sync.
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+TBD from developer
+(TODO/DELL)
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands for User Management
+Users may be managed via Linux tools like `useradd`, `usermod`, `passwd`, etc. They may also be managed via configuration.
+
+##### username
+`username <name> password <password-string> role <role-string>` -- Configures a user on the system with a given name, password, and role.
+* **name** is a text string of 1-32 alphanumeric characters
+* **password-string** is a text string of 1-32 alphanumeric characters
+* **role-string** is a text string consisting of a role name. In the initial release, the user is recommended to use "admin" and "operator" roles, as other roles will not be supported. A text string is desired instead of keywords so that in the future, more roles may be implemented and expanded.
+* Configuring another a user with the same **name** should result in modification of the existing user.
+
+`no username <name>` -- Deletes a user from the system.
+* **name** is a text string of 1-32 alphanumeric characters
+
+#### 3.6.2.2 Show Commands
```

Comment:
We should implement a command that displays the set of configured users (per tenant?)